In [1]:
import sys

In [2]:
L =[]
sys.getsizeof(L)

56

In [15]:
L.append(21248)

In [16]:
L

['Hello', 21, 212, 2124, 21248]

In [22]:
sys.getsizeof(L)

56

In [21]:
L.clear()

In [23]:
for i in range(1,50):
    L.append(i)
    print(sys.getsizeof(L))

88
88
88
88
120
120
120
120
184
184
184
184
184
184
184
184
248
248
248
248
248
248
248
248
312
312
312
312
312
312
312
312
376
376
376
376
376
376
376
376
472
472
472
472
472
472
472
472
472


By 32 the size of list gets extended. for empty list size was 56 and added element size increased to 88 and after entering 5 element size increased to 120....

He we are going to create our own List using the ctype module in python.
In python ctype isa  powerful tool that allows us to interact with C data types, and it provides a way of creating new
In Python, the ctypes module provides C compatible data types, and allows calling functions in DLLs/shared libraries. It can be used to wrap these libraries in Pure Python

In [25]:
import ctypes

In [26]:
class MyList():
    def __init__(self):
        self.size=1 # Will give how much item can be stored in the array
        self.n=0   # Will keep track of number of items  added so far
        #Create a C type array with size = self.size
        self.A=self._make_array(self.size)

    def _make_array(self,capacity):
        return (capacity*ctypes.py_object)()


In [42]:
L=MyList()

In [43]:
type(L)

__main__.MyList

In [44]:
len(L)

TypeError: object of type 'MyList' has no len()